<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1294</width>
    <height>747</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>KP-Scanner-SQLI</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1301</width>
      <height>761</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: #212121;</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QFrame" name="frame_2">
     <property name="geometry">
      <rect>
       <x>-10</x>
       <y>0</y>
       <width>1311</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #000000;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>10</y>
        <width>201</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI</family>
        <pointsize>12</pointsize>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>KP-Scanner-SQLI</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>760</x>
        <y>10</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap>image/banner.jpg</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QTabWidget" name="tabWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>170</y>
       <width>1291</width>
       <height>571</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Log</string>
      </attribute>
      <widget class="QListWidget" name="listscan">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>571</width>
         <height>501</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: #263238;
color: #00e5ff;
font-family: 'Roboto Mono';
font-size: 15px;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>71</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>Total Link:</string>
       </property>
      </widget>
      <widget class="QLabel" name="lb_total_link">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>10</y>
         <width>16</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
      <widget class="QListWidget" name="listheader">
       <property name="geometry">
        <rect>
         <x>590</x>
         <y>40</y>
         <width>671</width>
         <height>501</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: #263238;
color: rgb(255, 255, 0);
font-family: 'Roboto Mono';
font-size: 15px;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>590</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>Random Header</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Links</string>
      </attribute>
      <widget class="QTableWidget" name="tablelinks">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>1281</width>
         <height>521</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: #263238;
color: #00e5ff;
font-family: 'Roboto Mono';
font-size: 15px;</string>
       </property>
       <column>
        <property name="text">
         <string>URL</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Status</string>
        </property>
       </column>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Vulnerable</string>
      </attribute>
      <widget class="QTableWidget" name="tableVuln">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>1291</width>
         <height>241</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: #263238;
color: #00e5ff;
font-family: 'Roboto Mono';
font-size: 12px;</string>
       </property>
       <column>
        <property name="text">
         <string>Link</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Parameter</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Payload</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Signature</string>
        </property>
       </column>
      </widget>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>240</y>
         <width>1281</width>
         <height>311</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="title">
        <string>Detail</string>
       </property>
       <widget class="QFrame" name="frame_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>1261</width>
          <height>271</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <widget class="QFrame" name="frame_4">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>1281</width>
           <height>41</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">border-top-color: rgb(255, 255, 255);
border-right-color: rgb(255, 255, 255);
border-left-color: rgb(255, 255, 255);
border-bottom-color: rgb(255, 0, 0);
border-style: solid;
border-width:5px;</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <widget class="QLabel" name="label_7">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>121</width>
            <height>31</height>
           </rect>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(0, 0, 0);
border: none;</string>
          </property>
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:700;&quot;&gt;SQL Injection&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>50</y>
           <width>151</width>
           <height>171</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">color: rgb(0, 0, 0);</string>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;PCI 3.2 &lt;/span&gt;6.5.1&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;OWASP 2013 &lt;/span&gt;A1&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;OWASP 2017 &lt;/span&gt;A1&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;OWASP 2021&lt;/span&gt; A1&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;CVSS 9.0 SCORE&lt;/span&gt;&lt;br/&gt;&lt;span style=&quot; font-size:large; font-weight:700; color:#3e3e3e; background-color:#ffffff;&quot;&gt;Severities:&lt;/span&gt;&lt;span style=&quot; font-family:'consolas','lucida console','courier new','monospace'; font-size:12px; color:#222222; background-color:#ffffff;&quot;&gt;Critical&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_9">
         <property name="geometry">
          <rect>
           <x>140</x>
           <y>190</y>
           <width>21</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap>image/c.png</pixmap>
         </property>
        </widget>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>60</y>
           <width>81</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Lato','Helvetica Neue','Helvetica','Arial','sans-serif'; font-size:14px; font-weight:700; color:#454e59; background-color:#ffffff;&quot;&gt;Description &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
        <widget class="QPlainTextEdit" name="plainTextEdit">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>80</y>
           <width>211</width>
           <height>181</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">color: rgb(0, 0, 0);</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="plainText">
          <string>SQL injection testing checks if it is possible to inject data into the application so that it executes a user-controlled SQL query in the database. Testers find a SQL injection vulnerability if the application uses user input to create SQL queries without proper input validation. A successful exploitation of this class of vulnerability allows an unauthorized user to access or manipulate data in the database.
An SQL injection attack consists of insertion or “injection” of either a partial or complete SQL query via the data input or transmitted from the client (browser) to the web application. A successful SQL injection attack can read sensitive data from the database, modify database data (insert/update/delete), execute administration operations on the database (such as shutdown the DBMS), recover the content of a given file existing on the DBMS file system or write files into the file system, and, in some cases, issue commands to the operating system. SQL injection attacks are a type of injection attack, in which SQL commands are injected into data-plane input in order to affect the execution of predefined SQL commands.
In general the way web applications construct SQL statements involving SQL syntax written by the programmers is mixed with user-supplied data. Example:
select title, text from news where id=$id
In the example above the variable $id contains user-supplied data, while the remainder is the SQL static part supplied by the programmer; making the SQL statement dynamic.
Because the way it was constructed, the user can supply crafted input trying to make the original SQL statement execute further actions of the user’s choice. The example below illustrates the user-supplied data “10 or 1=1”, changing the logic of the SQL statement, modifying the WHERE clause adding a condition “or 1=1”.
select title, text from news where id=10 or 1=1
SQL Injection attacks can be divided into the following three classes:

Inband: data is extracted using the same channel that is used to inject the SQL code. This is the most straightforward kind of attack, in which the retrieved data is presented directly in the application web page.
Out-of-band: data is retrieved using a different channel (e.g., an email with the results of the query is generated and sent to the tester).
Inferential or Blind: there is no actual transfer of data, but the tester is able to reconstruct the information by sending particular requests and observing the resulting behavior of the DB Server.

A successful SQL Injection attack requires the attacker to craft a syntactically correct SQL Query. If the application returns an error message generated by an incorrect query, then it may be easier for an attacker to reconstruct the logic of the original query and, therefore, understand how to perform the injection correctly. However, if the application hides the error details, then the tester must be able to reverse engineer the logic of the original query.
About the techniques to exploit SQL injection flaws there are five commons techniques. Also those techniques sometimes can be used in a combined way (e.g. union operator and out-of-band):

Union Operator: can be used when the SQL injection flaw happens in a SELECT statement, making it possible to combine two queries into a single result or result set.
Boolean: use Boolean condition(s) to verify whether certain conditions are true or false.
Error based: this technique forces the database to generate an error, giving the attacker or tester information upon which to refine their injection.
Out-of-band: technique used to retrieve data using a different channel (e.g., make a HTTP connection to send the results to a web server).
Time delay: use database commands (e.g. sleep) to delay answers in conditional queries. It is useful when attacker doesn’t have some kind of answer (result, output, or error) from the application.</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>390</x>
           <y>60</y>
           <width>121</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Lato','Helvetica Neue','Helvetica','Arial','sans-serif'; font-size:14px; font-weight:700; color:#454e59; background-color:#ffffff;&quot;&gt;Recomendation &lt;/span&gt;&lt;span style=&quot; font-family:'Lato','Helvetica Neue','Helvetica','Arial','sans-serif'; font-size:14px; font-weight:700; color:#de815c; background-color:#ffffff;&quot;&gt;*&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
        <widget class="QPlainTextEdit" name="plainTextEdit_2">
         <property name="geometry">
          <rect>
           <x>390</x>
           <y>80</y>
           <width>231</width>
           <height>181</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">color: rgb(0, 0, 0);</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="plainText">
          <string>To secure the application from SQL injection vulnerabilities, refer to the SQL Injection Prevention CheatSheet.
To secure the SQL server, refer to the Database Security CheatSheet.

For generic input validation security, refer to the Input Validation CheatSheet.
Tools

SQL Injection Fuzz Strings (from wfuzz tool) - Fuzzdb
Francois Larouche: Multiple DBMS SQL Injection tool -SQL Power Injector
sqlbftools
Bernardo Damele A. G.: sqlmap, automatic SQL injection tool
icesurfer: SQL Server Takeover Tool - sqlninja
Muhaimin Dzulfakar: MySqloit, MySql Injection takeover tool
bsqlbf, a blind SQL injection tool in Perl

References

Top 10 2017-A1-Injection
SQL Injection

Technology specific Testing Guide pages have been created for the following DBMSs:

Oracle
MySQL
SQL Server
PostgreSQL
MS Access
NoSQL
ORM
Client Side

Whitepapers

Victor Chapela: “Advanced SQL Injection”
Chris Anley: “More Advanced SQL Injection”
David Litchfield: “Data-mining with SQL Injection and Inference”
Imperva: “Blinded SQL Injection”
Ferruh Mavituna: “SQL Injection Cheat Sheet”
Kevin Spett from SPI Dynamics: “SQL Injection”
Kevin Spett from SPI Dynamics: “Blind SQL Injection”
“ZeQ3uL” (Prathan Phongthiproek) and “Suphot Boonchamnan”: “Beyond SQLi: Obfuscate and Bypass”
Adi Kaploun and Eliran Goshen, Check Point Threat Intelligence &amp; Research Team: “The Latest SQL Injection Trends”

Documentation on SQL Injection Vulnerabilities in Products

Anatomy of the SQL injection in Drupal’s database comment filtering system SA-CORE-2015-003</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_12">
         <property name="geometry">
          <rect>
           <x>650</x>
           <y>60</y>
           <width>121</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-family:'Lato','Helvetica Neue','Helvetica','Arial','sans-serif'; font-size:14px; font-weight:700; color:#454e59; background-color:#ffffff;&quot;&gt;References&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
        <widget class="QPlainTextEdit" name="plainTextEdit_3">
         <property name="geometry">
          <rect>
           <x>650</x>
           <y>80</y>
           <width>251</width>
           <height>181</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">color: rgb(0, 0, 0);</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="plainText">
          <string>Top 10 2017-A1-Injection
SQL Injection

Technology specific Testing Guide pages have been created for the following DBMSs:

Oracle
MySQL
SQL Server
PostgreSQL
MS Access
NoSQL
ORM
Client Side

Whitepapers

Victor Chapela: “Advanced SQL Injection”
Chris Anley: “More Advanced SQL Injection”
David Litchfield: “Data-mining with SQL Injection and Inference”
Imperva: “Blinded SQL Injection”
Ferruh Mavituna: “SQL Injection Cheat Sheet”
Kevin Spett from SPI Dynamics: “SQL Injection”
Kevin Spett from SPI Dynamics: “Blind SQL Injection”
“ZeQ3uL” (Prathan Phongthiproek) and “Suphot Boonchamnan”: “Beyond SQLi: Obfuscate and Bypass”
Adi Kaploun and Eliran Goshen, Check Point Threat Intelligence &amp; Research Team: “The Latest SQL Injection Trends”

Documentation on SQL Injection Vulnerabilities in Products

Anatomy of the SQL injection in Drupal’s database comment filtering system SA-CORE-2015-003</string>
         </property>
        </widget>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>681</width>
       <height>111</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="title">
      <string>Dashboard</string>
     </property>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Target:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineurl">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>20</y>
        <width>311</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
     <widget class="QPushButton" name="btnscan">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>20</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Depth:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="linedepth">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>60</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>70</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Total Requests:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lb_total_requests">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>70</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>70</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Roboto</family>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Database:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lbdb">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>70</y>
        <width>191</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel{
	color: rgb(255, 255, 0);
}
</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>690</x>
       <y>50</y>
       <width>591</width>
       <height>111</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="title">
      <string>Sinh Viên</string>
     </property>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>30</y>
        <width>231</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;Trương Hữu Phúc 2033181058&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;Nguyễn Trung Kiên 2033181116&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
